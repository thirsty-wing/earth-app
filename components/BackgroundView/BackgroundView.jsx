import { View } from "react-native";
import { useSafeAreaInsets } from "react-native-safe-area-context";
import { theme } from "../../utils/theme";


/**
 * BackgroundView. Background View which covers the whole screen and contains
 * children in the safe area.
 *
 * @param {object} props
 * @param {JSX.Element[]} [props.children] children elements to render inside
 * @returns {JSX.Element}
 */
function BackgroundView(props) {

  const {
    children = [],
  } = props;

  const insets = useSafeAreaInsets();

  return(
    <View
      style={{
        // background fill the screen
        backgroundColor: theme.palette.background.primary,
        position: "absolute",
        top: 0,
        bottom: 0,
        left: 0,
        right: 0,

        // contain children in safe area
        paddingTop: insets.top,
        paddingBottom: insets.bottom,
        paddingLeft: insets.left,
        paddingRight: insets.right,
      }}
    >
      { children }
    </View>
  );
}

export default BackgroundView;
