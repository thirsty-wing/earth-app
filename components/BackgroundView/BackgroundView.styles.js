import { StyleSheet, Platform } from "react-native";
import { theme } from "../../utils/theme";

export const styles = StyleSheet.create({
  root: {
    backgroundColor: theme.palette.background.primary,
    position: "absolute",
    top: 0,
    bottom: 0,
    left: 0,
    right: 0,
    ...Platform.select({
      web: {
        paddingTop: theme.spacing(2),
      },
    }),
  },
});

export default styles;
