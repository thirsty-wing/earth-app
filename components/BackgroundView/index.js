import BackgroundView from "./BackgroundView";

export default BackgroundView;
