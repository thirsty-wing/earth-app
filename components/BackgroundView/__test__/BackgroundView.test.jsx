import BackgroundView from "../BackgroundView";
import { shallow } from "../../../utils/test/shallow";


jest.mock("react-native-safe-area-context", () => ({
    useSafeAreaInsets: () => ({
      left: 0,
      right: 0,
      top: 0,
      bottom: 0,
    }),
  })
);


it("should render", () => {
  const content = shallow(<BackgroundView/>);

  expect(content).toMatchSnapshot();
});

