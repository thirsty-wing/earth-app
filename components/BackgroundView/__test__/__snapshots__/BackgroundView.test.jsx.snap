// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<View
  style={
    {
      "backgroundColor": "#003",
      "bottom": 0,
      "left": 0,
      "paddingBottom": 0,
      "paddingLeft": 0,
      "paddingRight": 0,
      "paddingTop": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
/>
`;
